src/logPointsManager.ts

```diff
@@ -0,0 +1,280 @@
+    selectedDeploymentSlot: WebSiteModels.Site;
+    selectedInstance: WebSiteModels.SiteInstance;
+    sessionId: string;
+    processId: string;
+    debuggerId: string;
+
+    constructor(output: vscode.OutputChannel,
+        readonly azureAccount: AzureAccountWrapper,
+        readonly site: WebSiteModels.Site,
+        readonly subscription?: SubscriptionModels.Subscription
+    ) {
+        super(output);
+        if (util.isSiteDeploymentSlot(this.site)) {
+            this.selectedDeploymentSlot = this.site;
+        } else {
+            this.steps.push(new PromptSlotSelection(this, this.site));
+        }
+
+        this.steps.push(new GetUnoccupiedInstance(this));
+        this.steps.push(new PickProcessStep(this));
+        this.steps.push(new SessionAttachStep(this));
+        this.steps.push(new StartDebugAdapterStep(this));
+    }
+
+    get lastUsedPublishCredential(): WebSiteModels.User {
+        return this._cachedPublishCredential;
+    }
+
+    get webManagementClient() {
+        return new WebSiteManagementClient(this.azureAccount.getCredentialByTenantId(this.subscription.tenantId), this.subscription.subscriptionId);
+    }
+
+    async fetchPublishCrentential(site: WebSiteModels.Site): Promise<WebSiteModels.User> {
+        const siteClient = new WebSiteManagementClient(this.azureAccount.getCredentialByTenantId(this.subscription.tenantId), this.subscription.subscriptionId);
+        const user = await util.getWebAppPublishCredential(siteClient, site);
+        this._cachedPublishCredential = user;
+        return user;
+    }
+
+    async getCachedCredentialOrRefetch(site: WebSiteModels.Site): Promise<WebSiteModels.User> {
+        if (this.lastUsedPublishCredential) {
+            return Promise.resolve(this.lastUsedPublishCredential);
+        }
+
+        return this.fetchPublishCrentential(site);
+    }
+
+    protected beforeExecute() { }
+
+    protected onExecuteError(error: Error) {
+        if (error instanceof UserCancelledError) {
+            return;
+        }
+        this.writeline(`Deployment failed - ${error.message}`);
+        this.writeline('');
+    }
+}
+
+class PromptSlotSelection extends WizardStep {
+    constructor(private _wizard: LogPointsSessionAttach, readonly site: WebSiteModels.Site) {
+        super(_wizard, 'Choose a deployment slot.');
+    }
+
+    async execute(): Promise<void> {
+        let deploymentSlots: WebSiteModels.WebAppCollection;
+
+        // Decide if this AppService uses deployment slots
+        await vscode.window.withProgress({ location: vscode.ProgressLocation.Window }, async p => {
+            p.report({ message: 'Enumerating deployment slots for the App Service...' });
+            deploymentSlots = await this.getDeploymentSlots();
+        });
+
+        // if there is only one slot, just use that one and don't prompt for user selection.
+        if (deploymentSlots.length == 1) {
+            this._wizard.selectedDeploymentSlot = deploymentSlots[0];

```

Comment:
Investigate the behavior when an app service does not have a deployment slot.

Reply:
We have change the method to return both the deployment slots and the default production slot. So it will show at least 1.